firebase-ai/app/src/main/java/com/google/firebase/quickstart/ai/FirebaseAISamples.kt

```diff
@@ -339,4 +339,28 @@ val FIREBASE_AI_SAMPLES = listOf(
             )
         },
     ),
+    Sample(
+        title = "Imagen 3 - Server Template Generation",
+        description = "Generate an image using a server prompt template.",
+        navRoute = "imagen",
+        categories = listOf(Category.IMAGE),
+        initialPrompt = content { text("List of things that should be in the image") },
+        allowEmptyPrompt = false,
+        editingMode = EditingMode.TEMPLATE,
+        // To make this work on your project, create an `Imagen (Basic)` template in your project with this name
+        templateId = "imagen-basic",
+        templateKey = "prompt"
+    ),
+    Sample(
+        title = "Server Prompt Templates",
+        description = "Generate an invoice using server prompt templates",
+        navRoute = "text",
+        categories = listOf(Category.TEXT),
+        initialPrompt = content { text("Customer Name") },
```

Comment:
Should we pass a more realistic customer name?

```suggestion
        initialPrompt = content { text("Jane Doe") },
```